.github/workflows/ci_tests.yaml

```diff
@@ -0,0 +1,99 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
+
+name: Tests
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  # Schedule daily tests
+  schedule:
+    - cron: '0 0 * * *'
+
+jobs:
+  test:
+    name: ${{ matrix.os }} - Python ${{ matrix.python-version }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.7]
+        os: [ubuntu-latest, macOS-latest, windows-latest]
+
+    steps:
+      # Checkout current git repository
+      - name: Checkout
+        uses: actions/checkout@v2.3.1
+        with:
+          # fecth all history so that versioneer works
+          fetch-depth: 0
+
+      # Setup Miniconda
+      - name: Setup Miniconda
+        uses: goanpeca/setup-miniconda@v1.6.0
+        with:
+          python-version: ${{ matrix.python-version }}
+          channels: conda-forge
+
+      # Install GMT and other required dependencies from conda-forge
+      - name: Install GMT and required dependencies
+        shell: bash -l {0}
+        run: |
+          requirements_file=full-conda-requirements.txt
+          cat requirements.txt requirements-dev.txt > $requirements_file
+          cat << EOF >> $requirements_file
+          gmt=6.0.0
+          make
+          EOF
+          conda install --yes --file $requirements_file
```

Comment:
We can speed up the dependency install using caching too, see https://github.com/goanpeca/setup-miniconda#caching. I've used it for one of my projects, the relevant commit is at https://github.com/weiji14/deepicedrain/commit/391632bc590c43409be4913325df5f023708c5f3.

Reply:
Yes, I saw that. But I'm not sure how the caches are updated. For example, we may cache numpy v1.18.0 now. When new numpy versions are released, conda will download and install the latest version (e.g., numpy v1.19.0). What will happen to the caches? Do the caches contains the tarballs of numpy v1.18.0, v1.19.0 or both? Do we have to manually update the cache key?

Reply:
The cache would hash the `environment.yml/requirement.txt` files, and unless the hash changes, the cache will stay at v1.18.0 (unless we manually update the cache key). Maybe we should start thinking about a minimum supported dependency policy, as I suggested at https://github.com/GenericMappingTools/pygmt/issues/340#issuecomment-554545546.